### PR TITLE
Override -isEqual: as well as -isEqualToString:

### PR DESCRIPTION
**Pull request by Catfish-Man**

### CODE REVIEW

**Comment by Catfish-Man** [comment]
@swift-ci please smoke benchmark

File: stdlib/public/core/StringStorage.swift [hunk 1]
```diff
@@ -298,6 +298,12 @@ final internal class __StringStorage
   final internal func isEqual(to other: AnyObject?) -> Int8 {
     return _isEqual(other)
   }
+  
+  @objc(isEqual:)
+  @_effects(readonly)
+  final internal func isEqual(object other:AnyObject?) -> Int8 {
+    return _isEqual(other)
```
Comment (jrose-apple) [inline stdlib/public/core/StringStorage.swift]: Doesn't `_isEqual` assume the other object is an NSString already?

Comment (Catfish-Man) [reply stdlib/public/core/StringStorage.swift]: Nope. There's a comment in there about how we could do that according to the API contract, but are matching accidental behavior of NSString and not doing it.

**Comment by Catfish-Man** [comment]
I'm assuming the lack of wins here mostly boils down to not having tests that heavily exercise String-keyed NSDictionary lookup. I might try adding one of those.

Could also just be too small a win.

**Comment by Catfish-Man** [comment]
Interesting discovery: String as AnyObject uses this code path, but String as NSString doesn't

**Pull request closed without merging**